File: python/Case.py
from Komponen import Komponen

class Case(Komponen):
    def __init__(self, id="", warranty_months=0, merk="", nama="", harga=0.0,
                 form_factor="", supported_motherboards=None, fan_mounts=0, has_glass_panel=False):
        super().__init__(id, warranty_months, merk, nama, harga)
        self.form_factor = form_factor
        self.supported_motherboards = supported_motherboards if supported_motherboards is not None else []
        self.fan_mounts = fan_mounts
        self.has_glass_panel = has_glass_panel

    def set_form_factor(self, form_factor):
        self.form_factor = form_factor

    def set_supported_motherboards(self, supported_motherboards):
        self.supported_motherboards = supported_motherboards

    def add_supported_motherboard(self, motherboard_type):
        self.supported_motherboards.append(motherboard_type)

    def set_fan_mounts(self, fan_mounts):
        self.fan_mounts = fan_mounts

    def set_has_glass_panel(self, has_glass_panel):
        self.has_glass_panel = has_glass_panel

    def get_form_factor(self):
        return self.form_factor

    def get_supported_motherboards(self):
        return self.supported_motherboards

    def get_fan_mounts(self):
        return self.fan_mounts

    def get_has_glass_panel(self):
        return self.has_glass_panel

    def display_info(self):
        super().display_info()
        print(f"Case Specs: {self.form_factor}, Fan Mounts: {self.fan_mounts}")
        print(f"Glass Panel: {'Yes' if self.has_glass_panel else 'No'}")
        print("Supported Motherboards: ", end="")
        print(", ".join(self.supported_motherboards))

File: java/Case.java
import java.util.ArrayList;
import java.util.List;

public class Case extends Komponen {
    private String formFactor;
    private List<String> supportedMotherboards;
    private int fanMounts;
    private boolean hasGlassPanel;

    public Case() {
        super();
        this.formFactor = "";
        this.supportedMotherboards = new ArrayList<>();
        this.fanMounts = 0;
        this.hasGlassPanel = false;
    }

    public Case(String id, int warrantyMonths, String merk, String nama, double harga,
                String formFactor, List<String> supportedMotherboards, int fanMounts, boolean hasGlassPanel) {
        super(id, warrantyMonths, merk, nama, harga);
        this.formFactor = formFactor;
        this.supportedMotherboards = (supportedMotherboards != null) ? new ArrayList<>(supportedMotherboards) : new ArrayList<>();
        this.fanMounts = fanMounts;
        this.hasGlassPanel = hasGlassPanel;
    }

    public void setFormFactor(String formFactor) {
        this.formFactor = formFactor;
    }

    public void setSupportedMotherboards(List<String> supportedMotherboards) {
        this.supportedMotherboards = new ArrayList<>(supportedMotherboards);
    }

    public void addSupportedMotherboard(String motherboardType) {
        this.supportedMotherboards.add(motherboardType);
    }

    public void setFanMounts(int fanMounts) {
        this.fanMounts = fanMounts;
    }

    public void setHasGlassPanel(boolean hasGlassPanel) {
        this.hasGlassPanel = hasGlassPanel;
    }

    public String getFormFactor() {
        return formFactor;
    }

    public List<String> getSupportedMotherboards() {
        return supportedMotherboards;
    }

    public int getFanMounts() {
        return fanMounts;
    }

    public boolean getHasGlassPanel() {
        return hasGlassPanel;
    }

    public void displayInfo() {
        super.displayInfo();
        System.out.println("Case Specs: " + formFactor + ", Fan Mounts: " + fanMounts);
        System.out.println("Glass Panel: " + (hasGlassPanel ? "Yes" : "No"));
        System.out.print("Supported Motherboards: ");
        System.out.println(String.join(", ", supportedMotherboards));
    }
}
